udf/udf_get_key.sql

```diff
@@ -1,3 +1,18 @@
+/*
+
+Fetch the value associated with a given key from a "map type" imported from
+a Parquet dataset.
+
+Our machinery for importing datasets from Parquet models map types as a field
+containing a repeated struct named "key_value" with fields key and value.
```

Comment:
FWIW, I discovered in the course of looking into this that this is how parquet itself models the logical map type out of its primitive types: https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#maps